# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5], dtype=int64)

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5], dtype=int64)

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [22]:
# Определим долю задержек возвратов ('debt') для каждой группы по кол-ву детей ('children').
# Так как кол-во ункальных значений ко-ва детей не велико,
# то категоризацию по столбцу 'children' делать нет необходимости.

children_debt_mean = data.groupby('children').agg({'debt': ['count', 'mean']})
children_debt_mean.columns = children_debt_mean.columns.droplevel()
children_debt_mean.columns = ['count_clients', 'debt_mean_%']

# переводим доли в проценты и округляем значения до 2-х знаков
children_debt_mean['debt_mean_%']=children_debt_mean['debt_mean_%']*100
children_debt_mean = children_debt_mean.round(decimals=2)
children_debt_mean

count_clients  debt_mean_%
children                            
0                 14091         7.54
1                  4808         9.23
2                  2052         9.45
3                   330         8.18
4                    41         9.76
5                     9         0.00

In [23]:
# посчитаем средний процент задержек возврата для групп, имеющих от 1 до 4 детей.
# Группа заёмщиков с пятью детьми малочисленна и не может в данном случае приниматься во внимание.

with_children_mean = children_debt_mean.loc[1:4,'debt_mean_%' ].mean()
display ('{:.2f}' '%'.format(with_children_mean))

'9.15%'

**Вывод:**

Доля имевших задолжность среди клиентов без детей составляет 7.54%, что ниже, чем среднее значение для групп с детьми (9.16%).
Группа заёмщиков с пятью детьми малочисленна и не может в данном случае приниматься во внимание.

Сам Факт наличия детей, независимо от их количества, влияет на соблюдение сроков возврата кредитов.
Заёмщики с детьми имеют более высокую долю задеожек платежей. 

Влияние же именно количества детей на возврат кредита в срок не обнаружено.
Доли задержек в категориях, имеющих детей, находятся в диапазоне от 8,2% до 9,8% и не коррелируют с количеством детей в семье.



#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [24]:
# Группируем по 'family_status_id' и 'family_status', затем считаем количество клиентов и среднее значение 'debt' для каждого id
family_debt_mean = data.groupby(['family_status_id', 'family_status']).agg({'debt': ['count', 'mean']})

# Убираем двойную индексацию столбцов переименовав их
family_debt_mean.columns = ['count_clients', 'debt_mean_%']

# переводим доли в проценты, затем сортируем таблицу по возрастанию
family_debt_mean['debt_mean_%'] = family_debt_mean['debt_mean_%']*100
family_debt_mean = family_debt_mean.sort_values('debt_mean_%',ascending=True)

# округляем значения до 2-х знаков
family_debt_mean = family_debt_mean.round(decimals=2)
family_debt_mean

count_clients  debt_mean_%
family_status_id family_status                                    
2                вдовец / вдова                   951         6.62
3                в разводе                       1189         7.06
0                женат / замужем                12261         7.56
1                гражданский брак                4134         9.31
4                Не женат / не замужем           2796         9.76

**Вывод:**

Семейное положение влияет на возврат кредита в срок. 

При этом состоящие в гражданском браке, с точки зрения возврата кредитов, находятся на близком уровне с холостыми. Две эти группы наиболее рискованные для банка. Наблюдается заметный разрыв между этими двуми и остальными категориями.

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [25]:
# Группируем данные и считаем 'count' и среднее значение 'debt' для каждой категории
income_debt_mean = data.groupby('total_income_category').agg({'debt': ['count', 'mean']})

# Убираем двойную индексацию столбцов, затем переименовываем их
income_debt_mean.columns = income_debt_mean.columns.droplevel()
income_debt_mean.columns = ['count_clients', 'debt_mean_%']

# переводим доли в проценты, округляем значения до 2-х знаков
income_debt_mean['debt_mean_%'] = income_debt_mean['debt_mean_%']*100
income_debt_mean = income_debt_mean.round(decimals=2)

income_debt_mean

count_clients  debt_mean_%
total_income_category                            
A                                 25         8.00
B                               5014         7.06
C                              15921         8.50
D                                349         6.02
E                                 22         9.09

Категории от A до E расположены в таблице в порядке убывания дохода.
- A - наиболее богатые клиенты с доходом более 1 млн,
- B - доход от 200 тыс до 1 млн,
- C - доход от 50 до 200 тыс,
- D - доход от 30 до 50 тыс,
- E - наименее обеспеченные клиенты с доходом до 30 тыс.

Можно вилеть:
1. Категория E демонстрирует наихудший показатель возвратов, что кажется объяснимым;
2. с другой стороны, категория A также имеет высокий показатель нарушений, сопоставимый с категорией E;
3. наилучший показатель у категории D с доходами нижне-среднего диапазона, согласно имеющейся категоризации;
4. категория C, следующая по доходам за категорией D, демонстрирует резкое снижение надёжности заёмщиков.

**Вывод:** 

Из полученных результатов нельзя сделать однозначный вывод о влиянии уровня доход на надёжность заёмщзиков.
Похожие показатели у диаметрально противоположных категорий и резкие скачки при перехеде между соседними категориями,
могут быть результатом некорректной категоризации. 

>Сомнения в корректности применнённых для категоризации критериев подкрепляются в том числе, немногочисленностью крайних категорий - 22 и 25 клиентов из более чем 21 тыс.

**Рекомендации**

Проверить обоснованность критериев категоризации клоиентов по уровня дохода.



#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [26]:
# Определим долю задержек по возвратам ('debt') для каждой категории цели ('purpose_category').
purpose_debt_mean = data.groupby('purpose_category').agg({'debt': ['count', 'mean']})

# Убираем двойную индексацию столбцов, переименовав их
purpose_debt_mean.columns = ['count_clients', 'debt_mean_%']

# переводим доли в проценты, округляем значения до 2-х знаков и сортируем по возрастанию
purpose_debt_mean['debt_mean_%'] = purpose_debt_mean['debt_mean_%']*100
purpose_debt_mean = purpose_debt_mean.round(decimals=2)
purpose_debt_mean = purpose_debt_mean.sort_values('debt_mean_%',ascending=True)

purpose_debt_mean

count_clients  debt_mean_%
purpose_category                                    
операции с недвижимостью          10751         7.26
проведение свадьбы                 2313         7.91
получение образования              3988         9.25
операции с автомобилем             4279         9.35

**Вывод:**

Заметна зависимость задержек платежей от целей кредита.
Наименьшим риском обладают операции с недвихимостью (7,3%), наибольшим - опервции с автомобилем (9.3%).
>Данное положение вызывает интерес, так как кредиты в сфере недвижимости предоставляются на большие суммы и на более долгие сроки по сравнению с кредитами на операции с автомобилем и кажутся, на первый взгляд, более рискованными.
Но предоставленные данные опровергают такое суждение.
Целесообразно сегментировать операции с недвижимостью, разделив их на две категории:
>- а) некоммерческие операции с недвижимостью (частные),
>- б) коммерческие операции с недвижимостью.
Показатели в таких группах могут отличаться.

**Проверим возможность новой категоризации на основе имеющегося датасета.**


In [27]:
# Выводим уникальные значения 'purpose' из категории 'операции с недвижимостью'

purpose_realestate = (data[data['purpose_category']=='операции с недвижимостью'])
purpose_realestate = purpose_realestate['purpose'].unique()

purpose_realestate

array(['покупка жилья', 'операции с жильем', 'покупка жилья для семьи',
       'покупка недвижимости', 'покупка коммерческой недвижимости',
       'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'покупка своего жилья',
       'операции с недвижимостью', 'покупка жилья для сдачи',
       'ремонт жилью'], dtype=object)

**К сожалению, разделение операций с недвижимостью  на коммерческие и частные в данном случае невозможно.**

>Некоторые комментарий в столбце 'purpose' не позволяет однозначно дифференцировать цели.\
Например, такие формулировки, как: 'операции с жильем', 'покупка недвижимости', 'строительство жилой недвижимости', 'жилье' и др. не дают оснований для более детальной категоризаци.\
Кроме того, попавшая в категорию операций с недвижимостью, цель "ремонт жилья", может быть как сферой коммерческой деятельности, так и разновидностью потребительского кредита, без каких-либо признакой операций с недвижимостью вообще. 

**Рекомендации**

Представляется целесообразным рекомендовать банку разработать алгоритм более точной категоризации клиентов по целям кредита.



#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

**Ответ:**

Пропуски данных были обнаружены в столбцах days_employed и total_income.
Количество пропусков в столбцах одинаковое - 2174. (Согласно результату ранее сделанной проверки методами isna() и sum().)
Посчитаем для наглядности долю пропущенных значений:

In [28]:
# берём колическво значений из резулитатов ранее сделанной проверки методом info()
entries_total = 21525 
entries_days_empl = 19351
entries_total_inc = 19351

# так как количество пропущенных значений в обоих столбцах равны, то возтмём только один столбец
prop = 1-(entries_days_empl/entries_total)
print('{:.0%}'.format(prop))

10%


Всего пропущенных значений:

- в столбце days_employed - 2174, что составляет 10% от общего количества записей
- в столбце total_income также 2174 (10% записей)

Тот факт, что оба столбца - "Дни работы" и "Суммарный доход" - имеют равное количество пропусков, вряд ли является случайным совпадением. Вероятно, эти параметры связаны: либо алгоритмом рассчетов, либо алгоритмом ввода данных.

Параметр total_incom содержит дробное число с большим количеством знаков после запятой, и это говорит о том, что total_incom - расчетное значение, полученное, вероятно, на основе накопленных данных за определённый период. А значит, и отсутствие значений в таком случае представляется тоже результатом работы алгоритма системы (не ошибкой, не случайным пропуском, например).

**Наиболее вероятные причины пропусков:**
- а) показатели рассчитываются автоматически только после накопления данных за период не меньше установленного. Таким образом, эти 10 % - новые клиенты, данных о которых пока недостаточно для расчета показателя (допустим, нужны сведения за 6 мес, а есть пока только за 4); 
- б) данные вводятся путём сканирования бумажных бланков, и - в том случае, когда документ содержит критические ошибки, - данные не поступают в систему (например, расчетная ведомость от работодателя имеет нестандартную форму, с которой система банка не может работать).

>**Проверим эту гипотезу.**
Есть основания полагать, что показатели days_employed и total_income связаны между собой, и если NaN (отсутствие данных) в этих столбцах располагается в одних и тех же строках, то гипотеза будет верна: отсутствие данных зависит от
>- а) настройки системы, либо
>- б) некорректности исходников, загружаемых в систему;\
если в разных строках - от других факторов.

>Для проверки отфильтруем строки, где данные отсутствуют одновременно в двух столбцах, и посчитаем их количество. Если значение будет равно 2174, то есть разнице между полностью заполненными строками (21525) и строками с пропусками (19351), то гипотеза верна. Если это значение меньше (что говорит о наличии элементов, не попадающих в пересечение исследуемых множеств), то гипотеза не верна.


In [29]:
# Для проверки гипотезы проведём последовательную логическую фильтрацию и посчитаем её результат.
# Для подсчёта возьмём любой полностью заполненный столбец, например 'dob_years'.
# Обратимся к исходному датасету, сохранив его в отдельную переменную

try:
    df_hyp = pd.read_csv('/datasets/data.csv')
except:
    df_hyp = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

hypothesis = df_hyp[(df_hyp['days_employed'].isna()) & (df_hyp['total_income'].isna())]['dob_years'].count()
hypothesis

2174

**Вывод**

**Гипотеза верна** - пропуски вызваны: либо
- а) настройками системы,
- б) либо некорректностью первичных документов, загружаемых в систему техническими средствами.

В первом случае рекомендуется выяснить у компетентных сотрудников компании, насколько оправданы данные настройки.\
Во втором случае - проверить, как и из каких источников вносились данные, попавшие в исследуемый срез: что в них общего, что отличает их от других источников.


#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

**Ответ:**

Заполнить пропуски медианным значением — лучшее решение для количественных переменных, связанных с социально-экономическими явлениями.
>Медиана – значение, которое делит отсортированные по возрастанию данные на две равные части. То есть медиана показывает середину ряда данных.
Медиана почти не чувствительна к экстремальным значениям, находящимся на краях ряда, в отличие от среднего арифметического.

Медианное значение в большей степени учитывает вероятностное распределение значений в ряду, являясь "представителем" диапазона наиболее часто встречающихся значений. При заполнении пропусков медиана  подходит наилучшим образом, именно как наиболее вероятное значение.

### Шаг 4: общий вывод.

**ЗАДАЧА:** выявить факторы, оказывающие влияние на надёжность заёмщиков и оценить степень их влияния.

**ВЫОДЫ:**
1. Не все факторы, которые исследовались на основании предоставленных данных и предложенных методик, оказывают значимое влияние на поведение заёмщиков.
2. Некоторые факторы не удалось оценить с достаточной достоверностью. Требуется уточнение исходных данных, либо дополнительные исследования.

**Ниже приведены результаты исследований отдельно по каждому фактору.**

1. КОЛИЧЕСТВО ДЕТЕЙ

Сам факт наличия детей, независимо от их количества, влияет на соблюдение сроков возврата кредитов.\
Заёмщики без детей более надёжны по параметру задеожек платежей по кредитам, чем заёмщики с детьми.

Влияние же именно количества детей на возврат кредита в срок не обнаружено. 

>Доля имевших задолжность среди клиентов без детей составляет 7.54%, что ниже, чем среднее значение для групп с детьми (9.16%).
Группа заёмщиков с пятью детьми малочисленна и не может в данном случае приниматься во внимание.
Доли задержек в категориях, имеющих детей, находятся в диапазоне от 8,2% до 9,8% и не коррелируют с количеством детей в семье.

**Рекомендации**

Целесообразно сопоставить имеющиеся данные с данными общей статистики региона о составе семей и их доходах, чтобы оценить значимость группы с 5 детьми в статистике банка.
А также, сравнить данные из других отделений банка в регионе (если такие имеются), либо - разделить предоставленные данные на группы по отделениям, если предоставленные данные - общие для нескольких отделений банка (что нам в данном случае не известно), либо сравнить с данными других банков в регионе, если есть такая возможность.\
**Цель** - определить корректность исследуемых данных и подтвердить или опровергнуть вывод о том, что количество детей в семье не влияет на надёжность заёмщиков.

2. СЕМЕЙНОЕ ПОЛОЖЕНИЕ

Семейное положение влияет на возврат кредита в срок.

При этом состоящие в гражданском браке, с точки зрения надёжности, находятся на очень близком уровне с холостыми.\
Две эти группы наиболее рискованные для банка. Наблюдается заметный разрыв между этими двуми и остальными категориями.\
.


3. УРОВЕНЬ ДОХОДА

Из полученных результатов нельзя сделать однозначный вывод о влиянии уровня доход на надёжность заёмщзиков.\
Похожие показатели у диаметрально противоположных категорий и резкие скачки при перехеде между соседними категориями, могут быть результатом некорректной категоризации.

> **Обоснование**

> C одной стороны, категория E (нижний уровень доходов) демонстрирует наихудший показатель возвратов,\
с другой стороны, категория A (ниболее обеспеченные) также имеет высокий показатель нарушений, сопоставимый с категорией E.\
Наилучший показатель у категории D с доходами нижне-среднего диапазона, согласно имеющейся категоризации,\
однако, категория C, следующая по доходам непосредственно за категорией D, демонстрирует резкое снижение надёжности заёмщиков.

>Сомнения в корректности применнённых для категоризации критериев подкрепляются, в том числе, немногочисленностью этих крайних категорий - 22 и 25 клиентов из более чем 21 тысячи.

**Рекомендации**

Проверить обоснованность критериев категоризации клоиентов по уровню дохода. Желательно с привлечением не только экономистов, но и социологов и\или социальных психологов.  

4. ЦЕЛИ КРЕДИТА

Заметна зависимость задержек платежей от целей кредита.\
Наименьшим риском обладают операции с недвихимостью (7,3%), наибольшим - опервции с автомобилем (9.3%).

>Данное положение вызывает интерес, так как кредиты в сфере недвижимости предоставляются на большие суммы и на более долгие сроки по сравнению с кредитами на операции с автомобилем и поэтому кажутся более рискованными.\
Но предоставленные данные опровергают такое суждение.\
>Целесообразно, на наш взгляд, сегментировать операции с недвижимостью, разделив их на две категории:
>- некоммерческие операции с недвижимостью (частные),
>- колммерческие операции с недвижимостью. Показатели в таких группах могут отличаться.


**Рекомендации**

Представляется целесообразным рекомендовать банку разработать алгоритм более точной категоризации клиентов по целям кредита.

>Некоторые комментарий в столбце 'purpose' не позволяет однозначно дифференцировать цели.\
Например, такие формулировки, как: 'операции с жильем', 'покупка недвижимости', 'строительство жилой недвижимости', 'жилье' и др. не дают оснований для категоризаци.\
Кроме того, попавшая в категорию операций с недвижимостью, цель "ремонт жилья", может быть как сферой коммерческой деятельности, так и разновидностью потребительского кредита, без каких-либо свойств операций с недвижимостью вообще.

\
\
5.  ПРОБЛЕМА ПРОПУЩЕННЫХ ЗНАЧЕНИЙ В БАЗЕ ДАННЫХ

Доля пропущенных значений составляет:
- в столбце days_employed - 10%
- в столбце total_income - 10%.

Ниже приводим аиболее вероятные причины пропусков.\
Основания для выводов описаны в части, посвящённой этапам анализа. 

- **Настройки системы,**
вероятно данные рассчитываются автоматически только после накопления данных за период не меньше установленного. Таким образом, эти 10 % - новые клиенты, данных о которых пока недостаточно для расчета показателя (допустим, нужны сведения за 6 мес, а есть пока только за 4);

- **некорректностью первичных документов**, загружаемых в систему техническими средствами.
Возможно, данные вводятся путём сканирования бумажных бланков, и - в том случае, когда документ содержит критические ошибки, - данные не поступают в систему (например, расчетная ведомость от работодателя имеет нестандартную форму, с которой система банка не может работать).

**Рекомендации**

- рекомендуется выяснить у компетентных сотрудников компании, насколько оправданы именно такие настройки системы;
- проверить, как и из каких источников вносятся данные, попавшие в исследуемый срез: что в них общего, что отличает их от других источников.

